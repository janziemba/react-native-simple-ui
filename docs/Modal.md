# Modal

## Props
- `children` (_node_, **required**)
- `onOverlayPress` (_function_, default null)

## Examples

```jsx

```
