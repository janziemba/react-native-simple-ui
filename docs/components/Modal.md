---
id: modal
title: Modal
---
## Props
- `children` (_node_, **required**)
- `onOverlayPress` (_function_, default null)

<img src="../assets/Modal/1.png" alt="Modal" width="250" style="display: inline-block;">
