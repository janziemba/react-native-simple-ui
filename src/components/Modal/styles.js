// @flow

import { StyleSheet } from 'react-native';

import type { ThemeShapeType } from '../../types';

export type StylesType = {
    container: StyleSheet.Styles,
    overlay: StyleSheet.Styles,
};

export default ({ radiuses }: ThemeShapeType): StylesType => ({
    container: {
        backgroundColor: 'white',
        borderRadius: radiuses.small,
        width: '75%',
    },
    overlay: {
        ...StyleSheet.absoluteFillObject,
        alignItems: 'center',
        backgroundColor: 'rgba(0, 0, 0, 0.5)',
        flex: 1,
        justifyContent: 'center',
        zIndex: 2,
    },
});
