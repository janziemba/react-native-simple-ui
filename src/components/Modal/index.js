// @flow

import React, { PureComponent } from 'react';
import type { Node } from 'react';
import { TouchableWithoutFeedback, View } from 'react-native';

import withTheme from '../../themes/withTheme';
import Padding from '../Padding';
import styles from './styles';
import type { StylesType } from './styles';

type Props = {
    children: Node,
    onOverlayPress?: Function,
    styles: StylesType,
};

class Modal extends PureComponent<Props> {
    static defaultProps = {
        onOverlayPress: null,
    };
    render(): Node {
        const { children, onOverlayPress, styles } = this.props;

        return (
            <TouchableWithoutFeedback
                accessibilityRole="none"
                onPress={onOverlayPress}
            >
                <View style={styles.overlay}>
                    <View style={styles.container}>
                        <Padding size="huge">
                            {children}
                        </Padding>
                    </View>
                </View>
            </TouchableWithoutFeedback>
        );
    }
}

Modal = withTheme(styles, 'Modal')(Modal);

export default Modal;
