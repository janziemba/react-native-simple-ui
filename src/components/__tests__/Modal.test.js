import React from 'react';
import { View } from 'react-native';
import renderer from 'react-test-renderer';

import Modal from '../Modal';

it('renders correctly', () => {
    const tree = renderer.create(
        <Modal>
            <View />
        </Modal>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
