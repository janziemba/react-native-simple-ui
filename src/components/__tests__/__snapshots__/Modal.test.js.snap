// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.5)",
      "bottom": 0,
      "flex": 1,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 2,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 4,
        "width": "75%",
      }
    }
  >
    <View
      multiplier={1}
      size="huge"
      sizeBottom={null}
      sizeHorizontal={null}
      sizeLeft={null}
      sizeRight={null}
      sizeTop={null}
      sizeVertical={null}
      style={
        Object {
          "gigantic": Object {
            "padding": 24,
          },
          "huge": Object {
            "padding": 20,
          },
          "large": Object {
            "padding": 16,
          },
          "medium": Object {
            "padding": 12,
          },
          "padding": 20,
          "small": Object {
            "padding": 8,
          },
          "tiny": Object {
            "padding": 4,
          },
        }
      }
      styles={
        Object {
          "gigantic": Object {
            "padding": 24,
          },
          "huge": Object {
            "padding": 20,
          },
          "large": Object {
            "padding": 16,
          },
          "medium": Object {
            "padding": 12,
          },
          "padding": 20,
          "small": Object {
            "padding": 8,
          },
          "tiny": Object {
            "padding": 4,
          },
        }
      }
      theme={
        Object {
          "colors": Object {
            "layout": Object {
              "backgroundColor": "#ffffff",
            },
            "palette": Object {
              "danger": "#ff4136",
              "dark": "#484848",
              "disabled": "#999999",
              "info": "#75caeb",
              "light": "#f6f6f6",
              "primary": "#158cba",
              "success": "#28b62c",
              "warning": "#ff851b",
            },
            "text": Object {
              "disabled": "#999999",
              "placeholder": "#e2e4e6",
              "primary": "#3d3d3d",
              "secondary": "#9e9e9e",
            },
          },
          "components": Object {},
          "radiuses": Object {
            "gigantic": 16,
            "huge": 12,
            "large": 8,
            "medium": 6,
            "small": 4,
            "tiny": 2,
          },
          "spacing": Object {
            "gigantic": 24,
            "huge": 20,
            "large": 16,
            "medium": 12,
            "small": 8,
            "tiny": 4,
          },
          "typography": Object {
            "fontFamily": "System",
            "fontSizes": Object {
              "large": 16,
              "medium": 14,
              "small": 12,
            },
            "fontWeights": Object {
              "bold": "900",
              "medium": "800",
              "normal": "600",
            },
            "iconSet": "MaterialIcons",
            "sizes": Object {
              "h1": Object {
                "fontSize": 31,
                "fontWeight": "900",
              },
              "h2": Object {
                "fontSize": 25,
                "fontWeight": "900",
              },
              "h3": Object {
                "fontSize": 19,
                "fontWeight": "800",
              },
              "h4": Object {
                "fontSize": 17,
                "fontWeight": "800",
              },
              "h5": Object {
                "fontSize": 16,
                "fontWeight": "800",
              },
              "h6": Object {
                "fontSize": 15,
                "fontWeight": "700",
              },
              "normal": Object {
                "fontSize": 14,
                "fontWeight": "700",
              },
              "small": Object {
                "fontSize": 12,
              },
            },
          },
        }
      }
    >
      <View />
    </View>
  </View>
</View>
`;
